drivers/flash/flash_stm32f7x.c

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <flash.h>
+#include <init.h>
+#include <soc.h>
+
+#include <arch/cpu.h>
+#include <cortex_m/exc.h>
+
+#include "flash_stm32.h"
+
+#define STM32F7X_SECTOR_MASK		((u32_t) 0xFFFFFF07)
+
+bool flash_stm32_valid_range(struct device *dev, off_t offset, u32_t len,
+			     bool write)
+{
+	ARG_UNUSED(write);
+
+	return flash_stm32_range_exists(dev, offset, len);
+}
+
+static int write_byte(struct device *dev, off_t offset, u8_t val)
+{
+	struct stm32f7x_flash *regs = FLASH_STM32_REGS(dev);
+	int rc;
+
+	/* if the control register is locked, do not fail silently */
+	if (regs->cr & FLASH_CR_LOCK) {
+		return -EIO;
+	}
+
+	rc = flash_stm32_wait_flash_idle(dev);
+	if (rc < 0) {
+		return rc;
+	}
+
+	regs->cr &= ~CR_PSIZE_MASK;
+	regs->cr |= FLASH_PSIZE_BYTE;
+	regs->cr |= FLASH_CR_PG;
+
+	/* flush the register write */
+	__DSB();
+
+	*((u8_t *) offset + CONFIG_FLASH_BASE_ADDRESS) = val;
+
+	/* flush the register write */
+	__DSB();
+
+	rc = flash_stm32_wait_flash_idle(dev);
+	regs->cr &= (~FLASH_CR_PG);
+
+	return rc;
+}
+
+static int erase_sector(struct device *dev, u32_t sector)
+{
+	struct stm32f7x_flash *regs = FLASH_STM32_REGS(dev);
+	int rc;
+
+	/* if the control register is locked, do not fail silently */
+	if (regs->cr & FLASH_CR_LOCK) {
+		return -EIO;
+	}
+
+	rc = flash_stm32_wait_flash_idle(dev);
+	if (rc < 0) {
+		return rc;
+	}
+	if (sector > 11) {
+		sector += 4;
+	}
+
+	regs->cr &= STM32F7X_SECTOR_MASK;
+	regs->cr |= FLASH_CR_SER | (sector << FLASH_CR_SNB_Pos);
+	regs->cr |= FLASH_CR_STRT;
+
```

Comment:
Similarly to what is done for writing a byte, you should set the `PSIZE` field to `BYTE`. It's possible to get a bigger erase parallelism on the STM32F7, but it is not possible over the whole supported voltage range. Given we do not know the voltage, it should be forced to byte.

I believe this can be improved later by adding a new entry in the device tree, but that should be done later in an another patch.